daml/Marketplace/Distribution/Auction/Service.daml

```diff
@@ -54,11 +54,11 @@ template Service
         with
           createAuctionRequestCid : ContractId CreateAuctionRequest
         do
-          CreateAuctionRequest{..} <- fetchAndArchive createAuctionRequestCid
+          CreateAuctionRequest{request} <- fetchAndArchive createAuctionRequestCid
 
           -- Deposit the asset into the allocation account so they cannot be transferred/withdrawn without the signatore of the Auctioneer
-          depositCid <- exerciseByKey @AllocationAccountRule allocationAccount.id Allocation.Deposit with depositCid = assetDepositCid; ..
-          create Auction with status = Auction.Open; ..
+          depositCid <- exerciseByKey @AllocationAccountRule allocationAccount.id Allocation.Deposit with depositCid = request.depositCid
+          create Auction with status = Auction.Open; auctionId = request.auctionId; asset = request.asset, quotedAssetId = request.quotedAssetId; floorPrice = request.floorPrice; ..
```

Comment:
... so that here you can write `create request` that's defined above.

This way you don't need to re-list all the arguments, but you're not extracting-and-then-creating all the fields.

Does that make sense?

Reply:
I see that `status` is not part of the request, and I suspect that's why you haven't embedded the `request` as I suggest above.

Reply:
Yes, the status of the `Auction` shouldn't be provided by the client. 

Also, I don't want the template for the Auction to be defined by how its created. Ie, I don't want to embed the `CreateAuctionRequest` type from the `choice` into the `Auction` template.